### PR TITLE
Revised docs related to operator overloading

### PR DESCRIPTION
**Pull request by pkulikov**
This PR removes [How to: Use Operator Overloading to Create a Complex Number Class](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/how-to-use-operator-overloading-to-create-a-complex-number-class) article, following the discussion that started with [this comment](https://github.com/dotnet/docs/pull/7265#issuecomment-415863730)

Moreover, this PR updates the [operator keyword](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/operator) article with more information about how to overload an operator. Part of this information has previously resided in the [Overloadable operators](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/overloadable-operators) topic, which now just references the keyword topic.

### CODE REVIEW

File: docs/csharp/programming-guide/statements-expressions-operators/overloadable-operators.md [hunk 1]
```diff
@@ -1,54 +1,35 @@
...
+|[+=](../../language-reference/operators/addition-assignment-operator.md), [-=](../../language-reference/operators/subtraction-assignment-operator.md), [\*=](../../language-reference/operators/multiplication-assignment-operator.md), [/=](../../language-reference/operators/division-assignment-operator.md), [%=](../../language-reference/operators/modulus-assignment-operator.md), [&=](../../language-reference/operators/and-assignment-operator.md), [&#124;=](../../language-reference/operators/or-assignment-operator.md), [^=](../../language-reference/operators/xor-assignment-operator.md), [\<\<=](../../language-reference/operators/left-shift-assignment-operator.md), [>>=](../../language-reference/operators/right-shift-assignment-operator.md)|Assignment operators cannot be explicitly overloaded. However, when you overload a binary operator, the corresponding assignment operator, if any, is also implicitly overloaded. For example, `+=` is evaluated using `+`, which can be overloaded.|
```
Comment (pkulikov) [inline docs/csharp/programming-guide/statements-expressions-operators/overloadable-operators.md]: Rich diff makes the review of the changes made in this table easier.

File: docs/csharp/language-reference/keywords/operator.md [hunk 2]
```diff
@@ -12,9 +13,19 @@ ms.assetid: 59218cce-e90e-42f6-a6bb-30300981b86a
...
 
+To overload an operator on a custom class or struct, you create an operator declaration in the corresponding type. The operator declaration that overloads a built-in C# operator must satisfy the following rules:
+
+- It includes both a `public` and a `static` modifier.
+- It includes `operator X` where `X` is the name or symbol of the operator being overloaded.
+- Unary operators have one parameter, and binary operators have two parameters. In each case, at least one parameter must be the same type as the class or struct that declares the operator.
+
+For the information about how to define conversion operators, see the [explicit](explicit.md) and [implicit](implicit.md) keyword articles.
+
+For the overview of the C# operators that can be overloaded, see the [Overloadable operators](../../programming-guide/statements-expressions-operators/overloadable-operators.md) article.
```
Comment (rpetrusha) [inline docs/csharp/language-reference/keywords/operator.md]: - the overview --> an overview
- I think this is more effective if it's moved into the second bulleted list item. Though it's also fine here.

File: docs/csharp/programming-guide/statements-expressions-operators/overloadable-operators.md [hunk 3]
```diff
@@ -1,54 +1,35 @@
...
-```csharp
-public static Complex operator +(Complex c1, Complex c2) =>
-        new Complex(c1.real + c2.real, c1.imaginary + c2.imaginary);
-  
-// Override ToString() to display a complex number in the traditional format:
-public override string ToString() => $"{this.real} + {this.imaginary}i";
-```
-
-For more information, see [How to: Use Operator Overloading to Create a Complex Number Class](../../../csharp/programming-guide/statements-expressions-operators/how-to-use-operator-overloading-to-create-a-complex-number-class.md).
+For the information about how to overload an operator, see the [operator](../../language-reference/keywords/operator.md) keyword article.
```
Comment (rpetrusha) [inline docs/csharp/programming-guide/statements-expressions-operators/overloadable-operators.md]: the information --> information

File: docs/csharp/language-reference/keywords/operator.md [hunk 4]
```diff
@@ -12,9 +13,19 @@ ms.assetid: 59218cce-e90e-42f6-a6bb-30300981b86a
 
 Use the `operator` keyword to overload a built-in operator or to provide a user-defined conversion in a class or struct declaration.
 
+To overload an operator on a custom class or struct, you create an operator declaration in the corresponding type. The operator declaration that overloads a built-in C# operator must satisfy the following rules:
+
+- It includes both a `public` and a `static` modifier.
+- It includes `operator X` where `X` is the name or symbol of the operator being overloaded.
+- Unary operators have one parameter, and binary operators have two parameters. In each case, at least one parameter must be the same type as the class or struct that declares the operator.
+
+For the information about how to define conversion operators, see the [explicit](explicit.md) and [implicit](implicit.md) keyword articles.
```
Comment (rpetrusha) [inline docs/csharp/language-reference/keywords/operator.md]: the information --> information

**Review by rpetrusha — CHANGES_REQUESTED** [review]
This looks really good, @pkulikov; thanks for submitting the PR and eliminating the outdated topic. I've left a couple of minor comments.

**Review by rpetrusha — APPROVED** [review]
Thanks, @pkulikov, for making the additional changes. I'll merge your PR now.

**Pull request merged**